src/DateTime.php

```diff
@@ -218,31 +252,31 @@ public function modify($modify)
     /**
      * Returns the difference between two DateTime objects represented as a DateInterval.
      *
-     * @param \DateTime $datetime
+     * @param BaseDateTime $datetime
      * @param bool|false $absolute
+     *
+     * @throws InvalidArgumentException In case of $datetime is not a DateTime nor a oat\dtms\DateTime object.
+     *
      * @return DateInterval
-     * @throws \InvalidArgumentException In case of $datetime is not a DateTime nor a oat\dtms\DateTime object.
      */
-    public function diff($datetime, $absolute = false)
+    public function diff($datetime, $absolute = false): BaseDateInterval
     {
         $d1 = clone $this;
 
-        if ($datetime instanceof \DateTime) {
-            $d2 = new static($datetime->format(DateTime::ISO8601));
+        if ($datetime instanceof BaseDateTime) {
+            $d2 = new static($datetime->format(static::ISO8601));
         } elseif ($datetime instanceof DateTime) {
```

Comment:
suggestion: For me, this line would be a bit more readable (i.e. less confusing) if using `self` instead of the current class name (`DateTime`).

```suggestion
        } elseif ($datetime instanceof self) {
```

Reply:
I will keep it as it is for now just to be sure, that I will not break anything and will create a second PR with this change 🙂 
Thanks for the suggestion!